File: php-sdk/src/Models/Checklist.php
<?php

declare(strict_types=1);

namespace Cleanster\Models;

/** A checklist returned by the live checklist API. */
final class Checklist
{
    public readonly int    $id;
    public readonly bool   $isDefault;
    public readonly bool   $disabled;
    public readonly string $title;
    public readonly string $type;
    public readonly int    $totalTasks;
    public readonly int    $totalSubTasks;
    /** @var ChecklistTask[] */
    public readonly array  $tasks;
    /** Legacy response accessor. */
    public readonly string $name;
    /** @var ChecklistItem[] */
    public readonly array  $items;
    /** The original raw array returned by the API. */
    public readonly array  $raw;

    public function __construct(array $data)
    {
        $this->id            = (int)($data['id'] ?? 0);
        $this->isDefault     = (bool)($data['is_default'] ?? false);
        $this->disabled      = (bool)($data['disabled'] ?? false);
        $this->title         = (string)($data['title'] ?? '');
        $this->type          = (string)($data['type'] ?? '');
        $this->totalTasks    = (int)($data['totalTasks'] ?? 0);
        $this->totalSubTasks = (int)($data['totalSubTasks'] ?? 0);
        $this->tasks = array_map(
            fn(array $task) => new ChecklistTask($task),
            $data['tasks'] ?? []
        );
        $this->name  = (string)($data['name'] ?? $this->title);
        $this->items = array_map(
            fn(array $item) => new ChecklistItem($item),
            $data['items'] ?? []
        );
        $this->raw   = $data;
    }
}

File: csharp-sdk/src/Cleanster/Models/Checklist.cs
using System.Text.Json.Serialization;

namespace Cleanster.Models;

/// <summary>A reusable collection of cleaning tasks.</summary>
public sealed record Checklist
{
    [JsonPropertyName("id")] public int Id { get; init; }
    [JsonPropertyName("is_default")] public bool IsDefault { get; init; }
    [JsonPropertyName("disabled")] public bool Disabled { get; init; }
    [JsonPropertyName("title")] public string Title { get; init; } = "";
    [JsonPropertyName("type")] public string Type { get; init; } = "";
    [JsonPropertyName("totalTasks")] public int TotalTasks { get; init; }
    [JsonPropertyName("totalSubTasks")] public int TotalSubTasks { get; init; }
    [JsonPropertyName("tasks")] public List<ChecklistTask> Tasks { get; init; } = [];

    // Legacy response fields retained for source compatibility.
    [JsonPropertyName("name")] public string Name { get; init; } = "";
    [JsonPropertyName("items")] public List<ChecklistItem> Items { get; init; } = [];
}

/// <summary>A task, including its nested subtasks, in a reusable checklist.</summary>
public sealed record ChecklistTask
{
    [JsonPropertyName("image_name")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? ImageName { get; init; }
    [JsonPropertyName("title")] public string Title { get; init; } = "";
    [JsonPropertyName("totalSubtasks")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public int? TotalSubtasks { get; init; }
    [JsonPropertyName("subtasks")] public List<ChecklistSubtask> Subtasks { get; init; } = [];
}

/// <summary>A subtask in a reusable checklist task.</summary>
public sealed record ChecklistSubtask
{
    [JsonPropertyName("description")] public string Description { get; init; } = "";
    [JsonPropertyName("flag_request_photos")] public bool FlagRequestPhotos { get; init; }
    [JsonPropertyName("photos")] public List<string> Photos { get; init; } = [];
}

/// <summary>The structured checklist body accepted by create and update endpoints.</summary>
public sealed record CreateChecklistRequest
{
    [JsonPropertyName("title")] public string Title { get; init; } = "";
    [JsonPropertyName("tasks")] public List<ChecklistTask> Tasks { get; init; } = [];
}
